### PR TITLE
Fix snapshot argument for componentDidMount - Issue #985

### LINKED ISSUE #985: Enhancer.js doesn't pass snapshot argument in componentDidUpdate
In last React version (16.3.0), which is supported in current package.json, `componentDidUpdate` takes `snapshot` as third argument ([docs](https://reactjs.org/docs/react-component.html#componentdidupdate)). The current implementation of enhancer.js [doesn't consider it](https://github.com/FormidableLabs/radium/blob/master/src/enhancer.js#L254):
```jsx
    componentDidUpdate(prevProps, prevState) {
      if (super.componentDidUpdate) {
        super.componentDidUpdate.call(this, prevProps, prevState);
      } 
```

A potential fix looks simple. Would you be intested in a PR?

### PR DESCRIPTION
**Pull request by blainekasten**
Fixes #985 
-

### CODE REVIEW

File: src/__tests__/radium-test.js [hunk 1]
```diff
@@ -276,6 +276,40 @@ describe('Radium blackbox tests', () => {
...
+  it('passes snapshot to the componentDidUpdate of the component, Issue #985', () => {
+    const SNAPSHOT = 'SNAPSHOT';
+
+    @Radium
+    class SnapshotComp extends Component {
+      state = {
+        incrementToTriggerLifecycles: 0
+      };
+
+      componentDidUpdate(props, state, snapshot) {
```
Comment (ryan-roemer) [inline src/__tests__/radium-test.js]: Should we also assert this was actually called just to cover our bases?

Comment (blainekasten) [reply src/__tests__/radium-test.js]: Updated, and refactored a bit to be simpler.

File: src/__tests__/radium-test.js [hunk 2]
```diff
@@ -276,6 +276,36 @@ describe('Radium blackbox tests', () => {
...
+      }
+    }
+
+    sinon.spy(SnapshotComp.prototype, 'componentDidUpdate');
+
+    const TestComponent = Radium(SnapshotComp);
+
+    TestUtils.renderIntoDocument(<TestComponent />);
+
+    assert(SnapshotComp.prototype.componentDidUpdate.calledOnce);
```
Comment (ryan-roemer) [inline src/__tests__/radium-test.js]: Let's remove `assert` and replace with `sinon-chai` asserts which we use everywhere else.

So, something like:

```js
expect(SnapshotComp.prototype.componentDidUpdate).to.have.been.calledOnce;
// OR
expect(SnapshotComp.prototype.componentDidUpdate).to.have.callCount(1);
```

File: .eslintrc [hunk 3]
```diff
@@ -17,6 +17,7 @@ globals:
     require: true
     module: true
     expect: true
+    assert: true
```
Comment (ryan-roemer) [inline .eslintrc]: I don't think we need `assert` as the new place it was used is probably better served with a first-class `sinon-chai` assert :)

**Comment by ryan-roemer** [comment]
Weird other observation -- coverage seemed to have went way down... I haven't dug in but is there something up there that you can see?

https://github.com/FormidableLabs/radium/pull/1013#issuecomment-445480453

> Coverage decreased (-53.6%) to 41.667%

File: src/__tests__/radium-test.js (hunk 2 above)
Comment (blainekasten) [reply src/__tests__/radium-test.js]: done.

File: .eslintrc (hunk 3 above)
Comment (blainekasten) [reply .eslintrc]: done.

**Comment by blainekasten** [comment]
@ryan-roemer I'm hoping it's because I pushed in an `it.only` on accident. We'll find out soon! otherwise i have no idea how it would jump that much.

**Pull request merged**